ports/rp2/mpconfigport.h

```diff
@@ -71,8 +71,19 @@
 #define MICROPY_CONFIG_ROM_LEVEL                (MICROPY_CONFIG_ROM_LEVEL_EXTRA_FEATURES)
 #endif
 
+#ifndef MICROPY_HW_ENABLE_PSRAM
+#define MICROPY_HW_ENABLE_PSRAM (0)
+#endif
+
 // Memory allocation policies
+#if MICROPY_HW_ENABLE_PSRAM
+#define MICROPY_GC_STACK_ENTRY_TYPE             uint32_t
```

Comment:
Based on our discussion, how about adding the following here:

```
#ifndef MICROPY_ALLOC_GC_STACK_SIZE
#define MICROPY_ALLOC_GC_STACK_SIZE (1024) // Or some other number, 1024 is entirely arbitrary
#endif
```

Basically, if PSRAM is used on the RP2350, make the default GC stack size larger. Because PSRAM is currently only supported on the RP2350, the user will definitely have somewhere between ~0.5MB and ~16.5MB of RAM available; so we can safely increase the default, and mitigate the same headache for other users with lots of code and PSRAM.

I've tested this works to override the universal default of 64, but a board can still override it in the corresponding `mpconfigboard.h`